cpp/log/tree_signer-inl.h

```diff
@@ -178,23 +148,41 @@ util::Status TreeSigner<Logged>::SequenceNewEntries() {
               << util::ToBase64(pending_entry.Entry().Hash());
       continue;
     }
-
-    if (pending_entry.Entry().has_provisional_sequence_number()) {
-      // This is a recovery from a crashed sequencer run.
-      // Since the sequencing order is deterministic the assigned provisional
-      // sequence number should match our expectation:
-      CHECK_EQ(next_sequence_number,
-               pending_entry.Entry().provisional_sequence_number());
-    }
-    VLOG(0) << util::ToBase64(pending_entry.Entry().Hash()) << " = "
-            << next_sequence_number;
-    status = consistent_store_->AssignSequenceNumber(next_sequence_number,
-                                                     &pending_entry);
-    if (!status.ok()) {
-      return status;
+    const auto seq_it(sequenced_hashes.find(pending_hash));
+    if (seq_it == sequenced_hashes.end()) {
+      // Need to sequence this one.
+      VLOG(1) << util::ToBase64(pending_hash) << " = " << next_sequence_number;
+
+      // Record the sequence -> hash mapping
+      ct::SequenceMapping::Mapping* m(mapping.MutableEntry()->add_mapping());
+      m->set_sequence_number(next_sequence_number);
+      m->set_unsequenced_hash(pending_entry.Entry().Hash());
+      pending_entry.MutableEntry()->set_sequence_number(next_sequence_number);
+      ++num_sequenced;
+      ++next_sequence_number;
+    } else {
+      VLOG(1) << "Previously sequenced " << util::ToBase64(pending_hash)
+              << " = " << next_sequence_number;
+      pending_entry.MutableEntry()->set_sequence_number(seq_it->second);
```

Comment:
Umm... The updated entry is never written back to etcd? So, in etcd, the ```sequence_number``` field is always unset, it's all about the mapping?